pkg/workload/workload.go

```diff
@@ -647,7 +650,7 @@ func NeedsSecondPass(w *kueue.Workload) bool {
 		HasTopologyAssignmentsPending(w) &&
 		!IsAdmitted(w) &&
 		!IsFinished(w) &&
-		!IsEvicted(w)
+		!IsEvicted(w) || IsAdmitted(w) && HasFailedNodeAnnotation(w) && !IsFinished(w)
```

Comment:
This check for `!IsFinished(w)` seems redundant

Reply:
Done